src/main/webapp/app/shared/modal/AddMutationModal.tsx

```diff
@@ -80,507 +82,747 @@ function AddMutationModalContent({
-          Var_Residues: excludingAlteration.entity?.variantResidues || '',
-          Consequence: excludingAlteration.entity?.consequence?.name || '',
-          Excluding: [],
-          warning: excludingAlteration.warning ? excludingAlteration.message : null,
-          error: excludingAlteration.error ? excludingAlteration.message : null,
-        })),
-        warning: alteration.warning ? alteration.message : null,
-        error: alteration.error ? alteration.message : null,
-      };
+  function handleNormalAlterationChange(newValue: string, tabIndex: number) {
+    setTabStates(states => {
+      const newStates = _.cloneDeep(states);
+      newStates[tabIndex].alterationFieldValueWhileFetching = newValue;
+      return newStates;
     });
+    fetchNormalAlterationDebounced(newValue, tabIndex, tabStates);
+  }
 
-    function replaceOldMutations(oldMutationStates: MutationState[], newMutations: MutationState[], uuidToReplace: string) {
-      return oldMutationStates.reduce((accumulator: MutationState[], currentMutation) => {
-        if (uuidToReplace === currentMutation.uuid) {
-          return [...accumulator, ...newMutations];
-        }
-        return [...accumulator, currentMutation];
-      }, []);
+  async function fetchExcludedAlteration(
+    newAlteration: string,
+    tabIndex: number,
+    excludingIndex: number,
+    alterationData: AlterationData[]
+  ) {
+    const newParsedAlteration = parseAlterationName(newAlteration);
+
+    const currentState = alterationData[tabIndex];
+    const alteration = currentState.alteration.toLowerCase();
+    let excluding: string[] = [];
+    for (let i = 0; i < currentState.excluding.length; i++) {
+      if (i === excludingIndex) {
+        excluding.push(...newParsedAlteration.map(alt => alt.alteration.toLowerCase()));
+      } else {
+        excluding.push(currentState.excluding[excludingIndex].alteration.toLowerCase());
+      }
+    }
+    excluding = excluding.sort();
+    if (
+      alterationData.some(
+        state =>
+          state.alteration.toLowerCase() === alteration &&
+          _.isEqual(state.excluding.map(ex => ex.alteration.toLowerCase()).sort(), excluding)
+      )
+    ) {
+      notifyError(new Error('Duplicate alteration(s) removed'));
+      setTabStates(states => {
+        const newStates = _.cloneDeep(states);
+        newStates[tabIndex].excluding.splice(excludingIndex, 1);
+        return newStates;
+      });
+      return;
     }
 
-    setMutationStates(prev => {
-      if (excludingUuid) {
-        const newMutationStates = _.cloneDeep(prev);
-        const parentState = newMutationStates.find(state => state.uuid === uuid);
-        newAlterations.forEach(newAlteration => (newAlteration.Comment = null));
-        parentState.Excluding = replaceOldMutations(parentState.Excluding, newAlterations, excludingUuid);
-        return newMutationStates;
-      }
+    const newComment = newParsedAlteration[0].comment;
```

Comment:
Is this needed? At line 378, we could just use `newParsedAlteration[index].comment`?

Reply:
You are correct. Either way would be correct since the only way multiple alterations would exist is if the alteration changed from `V600E` to `V600E/G (comment)`. Since the comment would apply to both, I chose to use the first comment, but both would be the same anyway.

Reply:
Changing it to just use index